zirgen/Dialect/BigInt/Precompiles.md

```diff
@@ -0,0 +1,106 @@
+# Creating a precompile using the BigInt2 system
+
+Creating a new accelerator using BigInt2 requires a series of related changes across files in two repositories. This is the process you should follow if you want to create a new bigint2 precompile and make it available within the zkVM.
+
+To keep the demo simple, let's imagine that we want to create a trivial accelerator which sums two 128-bit numbers, which we'll call "add128".
+
+
+First, create files under [`zirgen/circuit/bigint/`] which will hold the accelerator builder function. We might call them `add128.h` and `add128.cpp`, declaring and then defining a function like this:
```

Comment:
```suggestion
## Write a bigint program
First, create files under [`zirgen/circuit/bigint/`] which will hold the accelerator builder function. We might call them `add128.h` and `add128.cpp`, declaring and then defining a function like this:
```

Reply:
> accelerator builder function
Maybe we can just say "bigint program" here. What do you think @tzerrell?

Reply:
I'm personally leaning toward "bigint program" at the moment when I discuss things myself. I might also change accelerator to precompile
```suggestion
## Write a bigint program
First, create files under [`zirgen/circuit/bigint/`] which will hold the precompile builder function. We might call them `add128.h` and `add128.cpp`, declaring and then defining a function like this:
```